tests/unit/test_appdirs.py

```diff
@@ -300,3 +301,78 @@ def test_user_config_dir_linux_home_slash(self, monkeypatch):
         monkeypatch.setattr(sys, "platform", "linux2")
 
         assert appdirs.user_config_dir("pip") == "/.config/pip"
+
+
+class TestGetWinFolder:
+
+    def test_win_folder_ctypes(self, monkeypatch):
+        if sys.platform != 'win32':
+            return
+
+        @pretend.call_recorder
+        def _get_win_folder_from_registry(csidl_name):
+            return "return-value-from-registry"
+
+        @pretend.call_recorder
+        def _get_win_folder_with_ctypes(csidl_name):
+            return "return-value-from-ctypes"
+
+        monkeypatch.setitem(sys.modules, 'ctypes', types.ModuleType('ctypes'))
+        monkeypatch.setattr(
+            appdirs,
+            "_get_win_folder_from_registry",
+            _get_win_folder_from_registry,
```

Comment:
I was actually hoping that `winreg.OpenKey` and `winreg.QueryValueEx` could be called for real on Windows. Is there a blocker ?

Reply:
Not blocker but I don’t think it’s meaningful. Their return values might not be predictable (being essentially system configuration).